mysys/CMakeLists.txt

```diff
@@ -79,9 +79,14 @@ ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|amd64|i386|i686")
     ENDIF()
   ENDIF()
 ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
-  IF(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+  IF(CMAKE_C_COMPILER_ID MATCHES "Clang|GNU")
     include(CheckCXXSourceCompiles)
 
+    CHECK_C_COMPILER_FLAG(-march=armv8-a+crc+crypto HAVE_ARMV8_CRC_CRYPTO_MARCH)
```

Comment:
for reasons (unknown) its doing a CXX test  so this should be `CHECK_CXX_COMPILER_FLAG`